doc/pages/DEVELOPER.md

```diff
@@ -1,20 +1,331 @@
+    list of architectures. Supported values are:
+
+    *   `Auto`
+    *   `Kepler`, `Maxwell`, `Pascal`, `Volta`
+    *   `CODE`, `CODE(COMPUTE)`, `(COMPUTE)`
+
+    `Auto` will automatically detect the present CUDA-enabled GPU architectures
+    in the system. `Kepler`, `Maxwell`, `Pascal` and `Volta` will add flags for
+    all architectures of that particular NVIDIA GPU generation. `COMPUTE` and
+    `CODE` are placeholders that should be replaced with compute and code
+    numbers (e.g.  for `compute_70` and `sm_70` `COMPUTE` and `CODE` should be
+    replaced with `70`. Default is `Auto`.  For a more detailed explanation of
+    this option see the
+    [`ARCHITECTURES` specification list](https://github.com/ginkgo-project/CudaArchitectureSelector/blob/master/CudaArchitectureSelector.cmake#L58)
+    section in the documentation of the CudaArchitectureSelector CMake module.
+
+For example, to build everything (in debug mode), use:
+
+```cmake
+cmake  -G "Unix Makefiles" -H. -BDebug -DCMAKE_BUILD_TYPE=Debug -DGINKGO_DEVEL_TOOLS=ON \
+      -DGINKGO_BUILD_TESTS=ON -DGINKGO_BUILD_REFERENCE=ON -DGINKGO_BUILD_OMP=ON \
+	  -DGINKGO_BUILD_CUDA=ON 
+cmake --build Debug
+```
+
+NOTE: Ginkgo is known to work with the `Unix Makefiles` and `Ninja` based
+generators. Other CMake generators are untested.
+
+### Third party libraries and packages
+
+Ginkgo relies on third party packages in different cases. These third party
+packages can be turned off by disabling the relevant options.
+
++ GINKGO_BUILD_CUDA=ON:
+  [CudaArchitectureSelector](https://github.com/ginkgo-project/CudaArchitectureSelector)
+  (CAS) is a CMake helper to manage CUDA architecture settings;
++ GINKGO_BUILD_TESTS=ON: Our tests are implemented with [Google
+  Test](https://github.com/google/googletest);
++ GINKGO_BUILD_BENCHMARKS=ON: For argument management we use
+  [gflags](https://github.com/gflags/gflags) and for JSON parsing we use
+  [RapidJSON](https://github.com/Tencent/rapidjson);
++ GINKGO_DEVEL_TOOLS=ON:
+  [git-cmake-format](https://github.com/gflegar/git-cmake-format) is our CMake
+  helper for code formatting.
+
+By default, Ginkgo uses the internal version of each package. For each of the
+packages `GTEST`, `GFLAGS` and `RAPIDJSON` and `CAS`, it is possible to force
+Ginkgo to try to use an external version of a package. For this, set the CMake
+option `-DGINKGO_USE_EXTERNAL_<package>=ON`.### Installing Ginkgo
+
+### Installation
+
+To install Ginkgo into the specified folder, execute the following command in
+the build folder
+
+```sh
+make install
+```
+
+If the installation prefix (see `CMAKE_INSTALL_PREFIX`) is not writable for your
+user, e.g. when installing Ginkgo system-wide, it might be necessary to prefix
+the call with `sudo`.
+
+After the installation, CMake can find ginkgo with `find_package(Ginkgo)`.
+An example can be found in the [`test_install`](test_install/CMakeLists.txt).
+
```

Comment:
Since there is no official way supported by Github to include other markdown documents (yet), could you consider Generating this file inside CMake (or just provide a script for that somewhere) which allows us to keep all of this up to date in a easy fashion?

[Here](https://stackoverflow.com/questions/18438907/embedding-one-markdown-document-in-another) someone proposes a bit of perl which should do the trick.

Reply:
Basically, this would require using a link in the form `[../../Install.md]` and hopefully that perl one liner would replace this line with the actual content of the file.

Similarly to how you already generate the tests with perl, we could have a call to that script inside CMake to generate the actual file from a template file with the links inside it.